### PR TITLE
sdit-1661 court appearance create initial commit

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingMappingApiService.kt [hunk 1]
```diff
@@ -27,4 +34,29 @@ class CourtSentencingMappingApiService(@Qualifier("mappingApiWebClient") webClie
...
+      "/mapping/court-sentencing/court-appearances/nomis-court-appearance-id/{courtAppearanceId}",
+      courtAppearanceId,
+    )
+    .retrieve()
+    .awaitBodyOrNullWhenNotFound()
+
+  suspend fun createCourtAppearanceMapping(
+    mapping: CourtAppearanceAllMappingDto,
+  ): CreateMappingResult<CourtAppearanceAllMappingDto> {
+    return webClient.post()
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingMappingApiService.kt:49]: this is in addition to the base class version as more than one entity is mapped

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationService.kt [hunk 2]
```diff
@@ -70,9 +74,61 @@ class CourtSentencingSynchronisationService(
...
+          courtAppearanceId = event.courtAppearanceId,
+        )
+      mappingApiService.getCourtAppearanceOrNullByNomisId(event.courtAppearanceId)?.let { mapping ->
+        telemetryClient.trackEvent(
+          "court-appearance-synchronisation-created-ignored",
+          telemetry + ("dpsCourtAppearanceId" to mapping.dpsCourtAppearanceId),
+        )
+      } ?: let {
+        val dpsCaseId =
+          nomisCourtAppearance.caseId?.let { mappingApiService.getCourtCaseOrNullByNomisId(it)?.dpsCourtCaseId }
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationService.kt:99]: need to look up the court case mapping also as it is required for the appearance create

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationIntTest.kt [hunk 3]
```diff
@@ -725,6 +728,172 @@ class CourtSentencingSynchronisationIntTest : SqsIntegrationTestBase() {
...
+              check {
+                assertThat(it["offenderNo"]).isEqualTo("A3864DZ")
+                assertThat(it["nomisBookingId"]).isEqualTo(NOMIS_BOOKING_ID.toString())
+                assertThat(it["nomisCourtAppearanceId"]).isEqualTo(NOMIS_COURT_APPEARANCE_ID.toString())
+                assertThat(it["dpsCourtAppearanceId"]).isEqualTo(DPS_COURT_APPEARANCE_ID)
+              },
+              isNull(),
+            )
+          }
+          // will not create a court case in DPS
```
Comment (lbennett-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationIntTest.kt:891]: should be appearance not case in the comment here. Will correct this in next PR to add more tests

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/SynchronisationQueueService.kt [hunk 4]
```diff
@@ -42,4 +45,5 @@ class SynchronisationQueueService(
 data class InternalMessage<T>(
   val body: T,
   val telemetryAttributes: Map<String, String>,
+  val entityName: String? = null,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/SynchronisationQueueService.kt:48]: did you end up not using this?

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationService.kt (hunk 2 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/courtsentencing/CourtSentencingSynchronisationService.kt:99]: see  below

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/SynchronisationQueueService.kt (hunk 4 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/SynchronisationQueueService.kt:48]: yes - will remove

**Pull request merged**